src/primitives.jl

```diff
@@ -179,6 +181,30 @@ end
 dropnull(X::NullableVector) = X.values[!X.isnull]    # -> Vector
 
 """
+    dropnull!(X::AbstractVector)
+
+Remove null entries of `X` in-place and return the updated vector. This is a
+no-op if no nulls are present.
+
+    dropnull!(X::NullableVector)
+
+Remove null entries of `X` in-place and return a `Vector` view of `X.values`
+"""
+function dropnull!(X::AbstractVector)
+    if !(Nullable <: eltype(X))
+        return X
+    end
+    deleteat!(X, find(isnull, X))
+end
+function dropnull!{T<:Nullable}(X::AbstractVector{T})
+    deleteat!(X, find(isnull, X))
+end
+# TODO: replace `find(X.isnull)` with `X.isnull` when
+# https://github.com/JuliaLang/julia/pull/20465 is merged and part of
+# current release (either v0.6 or v1.0)
+dropnull!(X::NullableVector) = deleteat!(X, find(X.isnull)).values
```

Comment:
Again, it should be more efficient to use the strategy I described at https://github.com/JuliaStats/NullableArrays.jl/pull/179#issuecomment-277437885.